crates/transaction-pool/src/blobstore/disk.rs

```diff
@@ -133,17 +135,12 @@ impl BlobStore for DiskFileBlobStore {
     ) -> Result<Vec<Option<BlobAndProofV1>>, BlobStoreError> {
         let mut result = vec![None; versioned_hashes.len()];
         for (_tx_hash, blob_sidecar) in self.inner.blob_cache.lock().iter() {
-            for (i, blob_versioned_hash) in blob_sidecar.versioned_hashes().enumerate() {
-                for (j, target_versioned_hash) in versioned_hashes.iter().enumerate() {
-                    if blob_versioned_hash == *target_versioned_hash {
-                        result[j].get_or_insert_with(|| BlobAndProofV1 {
-                            blob: Box::new(blob_sidecar.blobs[i]),
-                            proof: blob_sidecar.proofs[i],
-                        });
-                    }
+            let matches = match_versioned_hashes(blob_sidecar, versioned_hashes);
+            for (i, match_result) in matches.into_iter().enumerate() {
```

Comment:
this is a bit problematic because this now allocates `let mut result = vec![None; versioned_hashes.len()];` per `blob_sidecar`

we can solve this differently:

https://github.com/alloy-rs/alloy/pull/1882#discussion_r1902978189